src/log.cpp

```diff
@@ -10,22 +10,44 @@
 
 Log * Log::mLog;
 
-Log::Log()
-{
-}
 
-Log::~Log()
-{
+void Log::setLogOutput( LogOutput output, const QString& path ) {
+    Log * self = g();
+    switch (output) {
+    case LogOutput::Stderr:
+        self->mOutputFile = stderr;
+        break;
+    case LogOutput::Stdout:
+        self->mOutputFile = stdout;
+        break;
+    case LogOutput::Window:
+        self->mOutputFile = nullptr;
+        break;
+    case LogOutput::File:
+#ifdef    Q_OS_WIN
+        ::fopen_s(&self->mOutputFile, path.toLocal8Bit().constData(), "wt");
```

Comment:
1. Need to check the fopen_s return value. It could set mOutputFile to nonzero and still fail; the function doc doesn't say it guarantees fp not to change unless the file actually opens.
2. https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/fopen-s-wfopen-s?view=vs-2019 has unicode support, which is more reliable than local 8 bit; I suggest using Unicode (8/16) and passing the appropriate encoding.

Unicode support is relevant here, as many usernames on non-English Windows systems will have non-English characters and paths.

Reply:
Another - and IMHO better - option would be to use QFile, which handles all that internally. You can use it for streams too. See https://stackoverflow.com/questions/4453497/how-to-get-qiodevice-instance-for-stdin-stdout-stderr-text-streams-in-qtjambi#4453827 - this should be portable.

Reply:
I've refactored it with QFile. However I'm not sure if it would be better to use here `unique_ptr<QFile>` rather than raw pointer or it does not align well with Qt/project code style.

Reply:
No, if you're using QFile, no need to use a pointer. With FILE you have to use a pointer though, as its an opaque type.